firefox-ios/Client/Coordinators/Launch/LaunchCoordinator.swift

```diff
@@ -215,6 +215,10 @@ final class LaunchCoordinator: BaseCoordinator,
             manager.setAccepted(acceptedDate: acceptedDate)
             TermsOfServiceTelemetry().termsOfServiceAcceptButtonTapped(acceptedDate: acceptedDate)
 
+            // Refresh Nimbus enrollment after ToU acceptance
+            Experiments.shared.fetchExperiments()
+            _ = Experiments.shared.applyPendingExperiments()
+
```

Comment:
fetchExperiments is an async call and you should not be expecting to use the results here immediately.

Additionally, we are now changing nimbus behaviour to call applyPendingExperiments on first run, which is not the current behaviour.

We initialize Nimbus [here](https://github.com/mozilla-mobile/firefox-ios/blob/7ed961c62a525a54b77f6372040a705cc56a5c5d/MozillaRustComponents/Sources/MozillaRustComponentsWrapper/Nimbus/NimbusBuilder.swift#L239) . If we need TOS acceptance to properly enroll in Nimbus onboarding experiments, we will need to process Nimbus first startup (ie applyLocalExpeirments) *AFTER* TOU onboarding.